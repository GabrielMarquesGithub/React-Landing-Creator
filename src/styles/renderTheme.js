import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
