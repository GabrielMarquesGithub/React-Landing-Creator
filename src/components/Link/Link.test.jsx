import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Link } from '.';

describe('<Link />', () => {
  it('should render a Link', () => {
    renderTheme(<Link link="http://localhost:3000">Children</Link>);
    const link = screen.getByRole('link', { name: 'Children' });

    expect(link).toHaveAttribute('target', '_self');
  });

  it('should render a Link with correct atribute', () => {
    renderTheme(
      <Link link="http://localhost:3000" newTab={true}>
        Children
      </Link>,
    );
    const link = screen.getByRole('link', { name: 'Children' });

    expect(link).toHaveAttribute('target', '_blank');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Link link="http://localhost:3000">Children</Link>,
    );
    expect(container).toMatchSnapshot();
  });
});
