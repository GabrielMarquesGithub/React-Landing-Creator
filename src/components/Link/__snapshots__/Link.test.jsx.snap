// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should match snapshot 1`] = `
<div>
  <div
    class="sc-gsnTZi jMsOSA"
  >
    <a
      class="sc-bczRLJ gqHZrE"
      href="http://localhost:3000"
      target="_self"
    >
      Children
    </a>
  </div>
</div>
`;
